source/matplot/core/axes_type.cpp

```diff
@@ -429,7 +429,15 @@ namespace matplot {
                                         std::string axis_name,
                                         bool minor_ticks = false) {
             // cb is the only axis we don't unset if tics are empty
-            if (ax.visible() && !ax.tick_values().empty()) {
+            // r-axis labels should still be handled even if axis is invisible since we use the grid
```

Comment:
Wouldn't it be more semantically correct if setting up the grid made it visible? Because this way the thing is as if both visible and invisible. The user will be literally seeing the thing and `visible()` would give him `false`.

Reply:
Well the `visible()` correctly refers to the axis itself (which is hidden) and not the grid, however for polar plots the library basically treats the grid as the axis and disables the axis by default (as just a single r-axis is rather unhelpful for a circular plot). 
Semantically it could make sense to hide gnuplot's notion of the r-axis from the user entirely and treat gnuplot's polar grid as the r-axis instead, but that sounds like a major refactor...


> The user will be literally seeing the thing and `visible()` would give him `false`.

This is already the case: `polarplot()` sets `r_axis.visible(false)` because it refers to the axis, and not to the visible grid which however easily might be mistaken for the r_axis...

Reply:
Maybe this illustration clarifies things:

Output of `example_polarplot_10` with the fix applied:
<img width="280" alt="polar_plot_rticks_good_img" src="https://github.com/alandefreitas/matplotplusplus/assets/100471420/d3b6a6af-443d-4f69-a720-d9ca13454cf2">

Output of `example_polarplot_10` with the addition of `ax->r_axis().visible(true);`, result independent of fix:
<img width="280" alt="polar_plot_rticks_bad_img" src="https://github.com/alandefreitas/matplotplusplus/assets/100471420/32f5aa05-f546-45d6-ba2f-e3009d2ff82e">


P.S.: The r-tick label overlaps the t-tick degree label presumably because of my broken HiDPI screen fractional scaling setup (fonts only scaling...). When running the example on my non-HiDPI laptop screen the labels do not overlap.